src/table/tables.jl

```diff
@@ -0,0 +1,159 @@
+#######################################
+# Iterator structures
+
+struct DTableRowIterator
+    d::DTable
+end
+
+struct DTableColumnIterator
+    d::DTable
+end
+
+struct DTablePartitionIterator
+    d::DTable
+end
+
+
+
+#######################################
+# DTable functions
+
+Tables.istable(table::DTable) = true
+Tables.rowaccess(table::DTable) = true
+Tables.rows(table::DTable) = DTableRowIterator(table)
+Tables.columnaccess(table::DTable) = true
+Tables.columns(table::DTable) = DTableColumnIterator(table)
+
+
+function Tables.schema(table::DTable)
+    if table.schema !== nothing
+        return table.schema
+    end
+    # Figure out schema
+    chunk_f = chunk -> begin
+        r = isnonempty(chunk)
+        (r, r ? Tables.schema(chunk) : nothing)
+    end
+    c_idx = 1
+    r = (false, nothing)
+    while !r[1]
+        r = fetch(Dagger.spawn(chunk_f, table.chunks[c_idx]))
+        c_idx += 1
+    end
+    # cache results
+    table.schema = r[2]
+end
+
+
+function Tables.columnnames(table::DTable)
+    s = Tables.schema(table)
+    s === nothing ? nothing : s.names
+end
+
+
+function _getcolumn(table::DTable, col::Union{Symbol, Int})
+    chunk_col = (_chunk, _col) -> Tables.getcolumn(_chunk, _col)
+    v = [Dagger.spawn(chunk_col, chunk, col) for chunk in table.chunks]
+    p = (_chunk) -> (; :_ => fetch(_chunk))
+    Tables.getcolumn(TableOperations.joinpartitions(Tables.partitioner(p, v)), :_)
+end
+
+
+Tables.getcolumn(table::DTable, col::Symbol) = _getcolumn(table, col)
+Tables.getcolumn(table::DTable, idx::Int) = _getcolumn(table, idx)
+
+
+
+#######################################
+# DTableRowIterator functions
+
+Tables.schema(table::DTableRowIterator) = Tables.schema(table.d)
+length(table::DTableRowIterator) = length(table.d)
+
+function Base.iterate(iter::DTableRowIterator; c_idx=0)
```

Comment:
Can you document what `c_idx` does here?

Reply:
I'm just going to redo these iterator kwargs. Was speedrunning the iterators and this let me write fewer functions, but it isn't elegant now that I look at it